op-devstack/dsl/supervisor.go

```diff
@@ -149,13 +149,26 @@ func (s *Supervisor) AdvancedL2Head(chainID eth.ChainID, delta uint64, lvl types
 	s.require.NoError(err)
 }
 
-func (s *Supervisor) AdvancedUnsafeHead(chainID eth.ChainID, delta uint64) {
+func (s *Supervisor) WaitForL2HeadToAdvanceTo(chainID eth.ChainID, lvl types.SafetyLevel, blockID eth.BlockID) {
+	err := retry.Do0(s.ctx, 120, &retry.FixedStrategy{Dur: 500 * time.Millisecond}, func() error {
+		chStatus := s.L2HeadBlockID(chainID, lvl)
+		s.log.Info("Supervisor view",
+			"chain", chainID, "label", lvl, "current", chStatus.Number, "target", blockID.Number)
+		if blockID != chStatus {
```

Comment:
you could fatally quit the test here if the chStatus block number is > blockID, because then the sync status has already passed the block you're waiting for.

Reply:
Good idea, I updated the error messages and quit out of the retry eagerly in various hard failure cases.